### PR TITLE
[fs] delete in batches, accept stdin

### PR DESCRIPTION
**Pull request by danking**
Some quality-of-life stuff for `hailtop.aiotools.delete`.

Without deleting in batches, I found it impossible to delete very large lists of files because we create too many asyncio tasks.

### CODE REVIEW

File: hail/python/hailtop/utils/utils.py [hunk 1]
```diff
@@ -111,12 +112,14 @@ def digits_needed(i: int) -> int:
     return 1 + digits_needed(i // 10)
 
 
-def grouped(n: int, ls: List[T]) -> Iterable[List[T]]:
+def grouped(n: int, ls: Iterable[T]) -> Iterable[List[T]]:
```
Comment (daniel-goldstein) [inline hail/python/hailtop/utils/utils.py:115]: Based on the above usage, shouldn't this be `Iterator`?

Comment (daniel-goldstein) [inline hail/python/hailtop/utils/utils.py:115]: Could you leave a comment to replace this with [itertools.batched](https://docs.python.org/3/library/itertools.html#itertools.batched) when we (one day hopefully) hit 3.12?

**Review by daniel-goldstein — CHANGES_REQUESTED** [review]

File: hail/python/hailtop/utils/utils.py (hunk 1 above)
Comment (danking) [reply hail/python/hailtop/utils/utils.py]: We use `grouped` in other places where we pass a `List`. Those would be type failures if this was `Iterator[T]`.

Comment (danking) [reply hail/python/hailtop/utils/utils.py]: done

**Review by daniel-goldstein — APPROVED** [review]

**Pull request merged**